src/components/SiteDetail.js

```diff
@@ -0,0 +1,37 @@
+import React, { Component } from 'react';
+import summary from '../apis/summary';
+class SiteDetail extends Component {
+  state = {
+    loadedSites: null
+  };
+
+  componentDidMount() {
+    this.loadSiteData();
+  }
+
+  componentDidUpdate() {
+    this.loadSiteData();
+  }
+
+  loadSiteData() {
+    if (this.props.match.params.id) {
+      if (
+        !this.state.loadedSites ||
+        (this.state.loadedSites && this.state.loadedSites.id !== this.props.match.params.id)
+      ) {
+        summary.get('/sites/' + this.props.match.params.id).then(response => {
+          this.setState({ loadedSites: response.data });
+        });
```

Comment:
One thing with this block is that it can be a lot more readable (with the same perf) using Async/Await. 

```js
async loadSiteData = () => {
  const { id: matchId}  = this.props.match.params

(...)
  const { data: loadedSites } = await summary.get(`/sites/${matchId}`)
  this.setState({ loadedSites })
(...)
}

```